### ISSUE TITLE
Read only SD card when using Seedvault

### ISSUE THREAD

**Issue by Kamul-PL** [stevesoltys/backup#760]
Hello,
I've been using Seedvault with LineageOS for a few days now and I noticed, that my external SD card is read only since a few days also. I'm not 100% sure, that guilty is Seedvault, but I've made some research and here are my observations:
- I use the external SD card as my backup destination
- when I reboot my phone, the SD card is becoming writable again
- some time after next Seedvault activation the SD card is read only again (for test purposes I use manually triggered backups)
- it seems there's not such an effect if I use internal (emulated) SD card as a place for backups

My SD card (128 GB) is quite new (about 1 month), and not used much, so the hardware shouldn't be the problem.
Does anyone have any ideas, what could that mean? Has anyone had a similar experience? Should I create an issue here?

**Comment by grote** [comment]
> Should I create an issue here?

You already created an issue. There's also discussions for questions.

> I noticed, that my external SD card is read only since a few days also.

Seedvault uses the Storage Access Framework (SAF) to access the SD card. This is a public system API that also other apps like the file manager use. Seedvault isn't setting anything to read-only intentionally.

> My SD card (128 GB) is quite new (about 1 month), and not used much, so the hardware shouldn't be the problem.

Can still be the problem. Did you try running `fsck` on it?

You can also disable seedvault and see if the problem continues. If not, try writing large amounts of data via the file manager to the card and see if this creates readonly state.

**Comment by Kamul-PL** [comment]
Thank you for your reply!

> You already created an issue. There's also discussions for questions.

It's hard to disagree. ;)

> > My SD card (128 GB) is quite new (about 1 month), and not used much, so the hardware shouldn't be the problem.
> 
> Can still be the problem. Did you try running `fsck` on it?

Of course, new doesn't necessarily mean working, but the probability should be much higher.
About `fsck` - didn't I need root for that? Unfortunately I unrooted my phone quite recently. ;)

> You can also disable seedvault and see if the problem continues. If not, try writing large amounts of data via the file manager to the card and see if this creates readonly state.

I'll make some tests and get back here.

**Comment by grote** [comment]
>  About fsck - didn't I need root for that? Unfortunately I unrooted my phone quite recently. ;)

It is an SD card. Can't you power off the phone, remove the card, run tests with a computer and then put it back?

**Comment by Kamul-PL** [comment]
True, sometimes the hardest thing is to come up with the most obvious solutions.
But on the other hand - it's enough to reboot the phone to temporarily solve the problem. So, disconnecting it from the phone and placing in computer's card reader probably will do the same.
